hdfs-scheduler/src/main/java/org/apache/mesos/hdfs/api/ClusterResource.java

```diff
@@ -0,0 +1,64 @@
+package org.apache.mesos.hdfs.api;
+
+import org.apache.mesos.hdfs.config.HdfsFrameworkConfig;
+import org.apache.mesos.hdfs.state.LiveState;
+import org.apache.mesos.hdfs.state.AcquisitionPhase;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import com.google.inject.Inject;
+
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.Response.Status;
+
+/**
+ * Cluster management API.
+ */
+@Path("cluster")
+public class ClusterResource {
+  private final Log log = LogFactory.getLog(ClusterResource.class);
+
+  private HdfsFrameworkConfig config;
```

Comment:
If we're going to update mesos-site.xml while running, we do make sure the changes get persisted to the appropriate places. This will also fix the problem that we need to configure mesos-site.xml at compile time not at run time for the changes to effect the executors. It is in use by executors. In fact the data directories the executors write to is specified there, i.e. mesos.hdfs.data.dir.

I completely understand that configuration is a separate problem, but this patch is putting configuration into an inconsistent state and if we do adjust the configuration while the code is running, we need to make sure it is persisted consistently for the scheduler and the executors. I'm worried about putting in a patch that will affect scheduler failover and the executors and will potentially create bugs that didn't previously exist.